sockeye/rerank.py

```diff
@@ -41,36 +41,69 @@ class Reranker:
     :param return_score: If True, also return the sentence-level score.
     """
 
-    def __init__(self, metric: str,
-                 return_score: bool = False) -> None:
-        if metric == C.RERANK_BLEU:
+    def __init__(self, metric: str, return_score: bool = False) -> None:
+        self.metric = metric
+        self.return_score = return_score
+
+        if self.metric == C.RERANK_BLEU:
             # "add-k" smoothing is the best-performing method implemented in
             # sacrebleu.  See "Method 2" results from Chen and Cherry
             # (http://aclweb.org/anthology/W14-3346)
             self.scoring_function = partial(sacrebleu.sentence_bleu, smooth_method='add-k')
-        elif metric == C.RERANK_CHRF:
+        elif self.metric == C.RERANK_CHRF:
             self.scoring_function = sacrebleu.sentence_chrf  # type: ignore
+        elif self.metric.startswith(C.RERANK_ISOMETRIC):
+            self.scoring_function = utils.compute_isometric_score  # type: ignore
         else:
             raise utils.SockeyeError("Scoring metric '%s' unknown. Choices are: %s" % (metric, C.RERANK_METRICS))
 
-        self.return_score = return_score
 
-    def rerank(self, hypotheses: Dict[str, Any], reference: str) -> Dict[str, Any]:
+    def rerank(self, hypotheses: Dict[str, Any], reference: str, source: str,
+               isometric_alpha: float = 0.5) -> Dict[str, Any]:
```

Comment:
would it make sense to move this (optinal) alpha parameter into the constructor of `Reranker` to keep the signature of the `rerank()` method the same?

Reply:
thanks! i have made the update according to your suggestion.